### PR TITLE
Google OAuth2 client and Spreadsheet bulk update

### LINKED ISSUE #9: Spreadsheet uploader- Bulk upload
Now we uploading the transactions to the Google Spreadsheet by

```js
transactionsArray.map((transaction) => mapKeys(transaction, (_value, key) => key.toLowerCase()))
  .forEach(async (transaction) => {
    await addRow(worksheet, transaction);
  });
```

This can throw this error:
> Uncaught (in promise) Error: HTTP error 429 (Too Many Requests)

So we need to replace the [`GoogleSpreadsheet.addRow`](https://github.com/theoephraim/node-google-spreadsheet#googlespreadsheetaddrowworksheet_id-new_row-callback) with [`GoogleSpreadsheet.bulkUpdateCells`](https://github.com/theoephraim/node-google-spreadsheet#googlespreadsheetbulkupdatecellscells-callback).

### PR DESCRIPTION
**Pull request by baruchiro**
- [x] Try using the [`electron-google-oauth2`](https://github.com/getstation/electron-google-oauth2) as package.
- [x] Bug: upload the data to an empty sheet, then upload again and see if the transactions count increased.
- [x] Update the `secret`- remove the `GOOGLE_SERVICE_PRIVATE_KEY` secret and add `GOOGLE_CLIENT_ID` and `GOOGLE_CLIENT_SECRET`.

Fix #9 
Validate #64 after merging

### CODE REVIEW

**Comment by baruchiro** [comment]
@Arielgordon123 Ready for review.
Please see my tasks in the PR description, I need to finish them so don't merge this PR. Let me do that when I ready.

**Comment by baruchiro** [comment]
I still can't use the original [`electron-google-oauth2`](https://github.com/getstation/electron-google-oauth2) package, see [this bug](https://github.com/getstation/electron-google-oauth2/issues/9).

**Comment by baruchiro** [comment]
@Arielgordon123 Ready

File: src/components/MainPage/Exporters/SpreadsheetExporter.vue [hunk 1]
```diff
@@ -61,24 +97,55 @@ export default {
...
+          statusMessage,
         );
+
+        this.saveExporterProperties({ name, properties: this.properties });
       } catch (error) {
-        this.$logger.error(error);
+        this.$logger.error(error.message, error);
         this.emitStatus(false, error.message);
       }
       this.loading = false;
```
Comment (Arielgordon123) [inline src/components/MainPage/Exporters/SpreadsheetExporter.vue]: I think this should be in finally block

File: src/components/MainPage/Exporters/SpreadsheetExporter.vue [hunk 2]
```diff
@@ -61,24 +97,55 @@ export default {
     async submitForm() {
       this.loading = true;
       try {
-        this.saveExporterProperties({ name, properties: this.properties });
+        if (this.isNewSpreadsheet) {
+          await this.createNewSpreadsheet();
+        }
+
         const result = await saveTransactionsToGoogleSheets(
```
Comment (Arielgordon123) [inline src/components/MainPage/Exporters/SpreadsheetExporter.vue]: what will happen if the spreadsheet exists but there isn't _isbd sheet on it?

File: src/components/MainPage/Exporters/SpreadsheetExporter.vue (hunk 1 above)
Comment (baruchiro) [reply src/components/MainPage/Exporters/SpreadsheetExporter.vue]: I think the only advantage of the `finally` block is to run even if the `try` block includes a `return` statement.

But for clean code, I think it cleaner to wrap all the code with `try-catch-finally`.

File: src/components/MainPage/Exporters/SpreadsheetExporter.vue (hunk 2 above)
Comment (baruchiro) [reply src/components/MainPage/Exporters/SpreadsheetExporter.vue]: I forgot to test that. I created a new method to create a sheet if it not exist, right before the upload.

**Pull request merged**